code/game/objects/items/devices/floor_painter.dm

```diff
@@ -0,0 +1,77 @@
+// Floor painter
+
+/obj/item/device/floor_painter
+	name = "floor painter"
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "labeler1"
+	item_state = "flight"
+	
+	var/floor_icon
+	var/floor_state = "floor"
+	var/floor_dir = SOUTH
+	
+	var/list/allowed_states = list("arrival", "arrivalcorner", "bar", "barber", "blackcorner", "blue", "bluecorner", 
```

Comment:
Oh, true, this will be defined on every instance of the painter, and since BYOND is naive as all heck, many floor painters will have many of these large lists